ext/js/dictionary/dictionary-data-util.js

```diff
@@ -288,23 +304,28 @@ export class DictionaryDataUtil {
 
     /**
      * @param {string} reading
-     * @param {number} position
-     * @param {number[]} nasalPositions
-     * @param {number[]} devoicePositions
-     * @param {import('dictionary').Tag[]} tags
+     * @param {import('dictionary').Pronunciation} pronunciation
      * @param {import('dictionary-data-util').GroupedPronunciationInternal[]} groupedPronunciationList
      * @returns {?import('dictionary-data-util').GroupedPronunciationInternal}
      */
-    static _findExistingGroupedPronunciation(reading, position, nasalPositions, devoicePositions, tags, groupedPronunciationList) {
-        for (const pitchInfo of groupedPronunciationList) {
+    static _findExistingGroupedPronunciation(reading, pronunciation, groupedPronunciationList) {
+        for (const groupedPronunciation of groupedPronunciationList) {
             if (
-                pitchInfo.reading === reading &&
-                pitchInfo.position === position &&
-                this._areArraysEqual(pitchInfo.nasalPositions, nasalPositions) &&
-                this._areArraysEqual(pitchInfo.devoicePositions, devoicePositions) &&
-                this._areTagListsEqual(pitchInfo.tags, tags)
+                groupedPronunciation.reading === reading &&
+                this._areTagListsEqual(groupedPronunciation.tags, pronunciation.tags) &&
+                (
+                    pronunciation.type === 'pitch-accent' &&
+                    groupedPronunciation.type === 'pitch-accent' &&
+                    groupedPronunciation.position === pronunciation.position &&
+                    this._areArraysEqual(groupedPronunciation.nasalPositions, pronunciation.nasalPositions) &&
+                    this._areArraysEqual(groupedPronunciation.devoicePositions, pronunciation.devoicePositions)
+                    ||
```

Comment:
&& and || precedence combination unclear and somewhat inefficient, create a helper function for this and do a `switch` on `.type`.